### ISSUE TITLE
Make grpc optional dependency for exporter

### ISSUE THREAD

**Issue by chulkilee** [open-telemetry/opentelemetry-erlang#179]
When an end user only want to export via OTLP/HTTP (either binary-encoded Protobuf or JSON Protobuf) - it will not use gRPC at all. In such cases, it would be better to avoid including the grpc dependency (`grpcbox`) if possible.

(low priority though)

**Comment by tsloughter** [comment]
Agreed, I'd wanted to do this and was hoping https://github.com/erlang/otp/pull/2675 would be merged a while ago. Hopefully it is merged for OTP 24.

**Comment by tsloughter** [comment]
That PR has been merged to OTP so this can be planned for OTP-24.

**Comment by jerryluk** [comment]
We just hit a new consequence of the hard `grpcbox` dependency that upgrades this from "avoidable bloat" to "breaks unrelated functionality", so adding it here as extra motivation.

**The problem:** hackney ≥ 4.0 implements HTTP/2 via the hex [`h2`](https://hex.pm/packages/h2) package. grpcbox's HTTP/2 engine is [`ts_chatterbox`](https://hex.pm/packages/ts_chatterbox). The two are related codebases and **define the same module names** (`h2_connection`, `h2_stream`, …). The BEAM has a single global module namespace, so only one of them loads — and when chatterbox wins the code-path race, every ALPN-negotiated HTTP/2 request through hackney crashes:

```
** (UndefinedFunctionError) function :h2_connection.start_link/4 is undefined or private
    (chatterbox 0.15.1) :h2_connection.start_link(:client, {:sslsocket, ...}, #PID<...>, %{})
    (hackney 4.5.2) src/hackney_conn.erl:2788: :hackney_conn.start_h2_connection/4
```

Because `opentelemetry_exporter` lists `grpcbox` in its OTP `applications`, chatterbox's modules are loaded in every app that depends on the exporter — **including apps that only ever use the default `http_protobuf` protocol and never execute a single grpcbox code path**. In our case (OTLP/HTTP to port 4318), adding no gRPC usage at all, requests to any h2-advertising server (Google APIs, most CDNs) crashed once we upgraded hackney.

Versions: `opentelemetry_exporter` 1.10.0 (also verified unchanged on `main`), `grpcbox` 0.17.1, `ts_chatterbox` 0.15.1, `hackney` 4.5.2, `h2` 0.10.4, OTP 29 / Elixir 1.20.

**Workaround** for anyone else landing here: pin hackney to HTTP/1.1 — `config :hackney, default_protocols: [:http1]` (Elixir) / `application:set_env(hackney, default_protocols, [http1])` — which forgoes HTTP/2 entirely.

On the fix: the `optional_applications` OTP support referenced above ([erlang/otp#2675](https://github.com/erlang/otp/pull/2675)) merged in OTP 24, and current exporter releases already require far newer OTP — so the original blocker seems long gone. Making `grpcbox` an `optional_applications` entry (plus marking the dependency optional in the package metadata, with a docs note to add `grpcbox` explicitly when using `otlp_protocol: grpc`) would let `http_protobuf` users drop grpcbox/chatterbox from their tree entirely and avoid the module collision.

Happy to test a branch against our setup if that helps.

**Comment by tsloughter** [comment]
@jerryluk thanks. I am making releases of otel packages to address this module collision but will also do the optional applications change. But the hex change isn't at this time possible because rebar3 doesn't support it.
